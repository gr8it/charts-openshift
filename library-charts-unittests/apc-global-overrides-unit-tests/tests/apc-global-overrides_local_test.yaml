suite: APC global overrides - local values only
templates:
  - templates/apc-global-overrides.yaml
values:
  - ../values.local.yaml
release:
  name: agout
  namespace: default

tests:
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}

  - it: customer from local
    asserts:
      - equal: { path: data.customer, value: customer-local }

  - it: repo fields from local
    asserts:
      - equal: { path: data.repoURL, value: repoURL-local }
      - equal: { path: data.repoTargetRevision, value: repoTargetRevision-local }

  - it: environment fields from local
    asserts:
      - equal: { path: data.environment, value: environment-local }
      - equal: { path: data.environmentShort, value: environmentShort-local }

  - it: cluster identity fields from local
    asserts:
      - equal: { path: data.clusterName, value: clusterName-local }
      - equal: { path: data.clusterAcmName, value: clusterAcmName-local }
      - equal: { path: data.clusterType, value: clusterType-local }

  - it: cluster domain and API fields from local
    asserts:
      - equal: { path: data.clusterBaseDomain, value: clusterBaseDomain-local }
      - equal: { path: data.clusterAppsDomain, value: clusterAppsDomain-local }
      - equal: { path: data.clusterApiURL, value: clusterApiURL-local }

  - it: cluster API versions from local
    asserts:
      - equal: { path: data.clusterKubeVersion, value: clusterKubeVersion-local }
      - equal:
          path: data.clusterApiVersions
          value:
            - clusterApiVersions-local

  - it: clusterServices and merged (local only, no global keys in non-merged)
    asserts:
      - equal: { path: data.clusterServices.local.url, value: clusterServicesLocalUrl-local }
      - equal: { path: data.clusterServices.override.url, value: clusterServicesOverrideUrl-local }
      - notExists: { path: data.clusterServices.global }
      - equal: { path: data.clusterServicesMerged.local.url, value: clusterServicesLocalUrl-local }
      - equal: { path: data.clusterServicesMerged.override.url, value: clusterServicesOverrideUrl-local }
      - notExists: { path: data.clusterServicesMerged.global }

  - it: booleans from local
    asserts:
      - equal: { path: data.clusterIsHub, value: false }
      - equal: { path: data.clusterRunsApps, value: false }

  - it: proxy, noProxy, proxyCIDRs from local
    asserts:
      - equal: { path: data.proxy, value: proxy-local }
      - equal: { path: data.noProxy, value: noProxy-local }
      - equal:
          path: data.proxyCIDRs
          value:
            - proxyIP-local1
            - proxyIP-local2

  - it: services and merged (local only, no global keys in non-merged)
    asserts:
      - equal: { path: data.services.local.url, value: servicesLocaltUrl-local }
      - notExists: { path: data.services.global }
      - equal: { path: data.servicesMerged.local.url, value: servicesLocaltUrl-local }
      - notExists: { path: data.servicesMerged.global }
      - equal: { path: data.servicesExtracted.certManager.defaultClusterIssuer, value: certManagerDefaultClusterIssuer-local }
      - equal: { path: data.servicesExtracted.crossplane.kubeVaultProviderConfigName, value: crossplaneKubeVaultProviderConfigName-local }
      - equal: { path: data.servicesExtracted.externalSecretsOperator.defaultClusterSecretStore, value: ESODefaultClusterSecretStore-local }
      - equal: { path: data.servicesExtracted.vault.kubeAuthMountPath, value: vaultKubeAuthMountPath-local }
      - equal: { path: data.servicesExtracted.vault.name, value: vaultName-local }
      - equal: { path: data.servicesExtracted.vault.url, value: vaultUrl-local }

  - it: caCertificates and merged (local only, no global keys in non-merged)
    asserts:
      - equal:
          path: data.caCertificates["override.crt"]
          value: |
            -----BEGIN CERTIFICATE-----
            local
            -----END CERTIFICATE-----
      - equal:
          path: data.caCertificates["local.crt"]
          value: |
            -----BEGIN CERTIFICATE-----
            local
            -----END CERTIFICATE-----
      - notExists:
          path: data.caCertificates["global.crt"]
      - equal:
          path: data.caCertificatesMerged["override.crt"]
          value: |
            -----BEGIN CERTIFICATE-----
            local
            -----END CERTIFICATE-----
      - equal:
          path: data.caCertificatesMerged["local.crt"]
          value: |
            -----BEGIN CERTIFICATE-----
            local
            -----END CERTIFICATE-----
