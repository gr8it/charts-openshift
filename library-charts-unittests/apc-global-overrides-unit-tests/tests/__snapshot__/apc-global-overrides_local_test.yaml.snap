manifest should match snapshot:
  1: |
    data:
      caCertificates:
        local.crt: |
          -----BEGIN CERTIFICATE-----
          local
          -----END CERTIFICATE-----
        override.crt: |
          -----BEGIN CERTIFICATE-----
          local
          -----END CERTIFICATE-----
      caCertificatesBundle: |
        -----BEGIN CERTIFICATE-----
        local
        -----END CERTIFICATE-----
        -----BEGIN CERTIFICATE-----
        local
        -----END CERTIFICATE-----
      caCertificatesBundleB64: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCmxvY2FsCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0KLS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCmxvY2FsCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0=
      caCertificatesBundleB64Safe: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCmxvY2FsCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0KLS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCmxvY2FsCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
      caCertificatesExtracted: ""
      caCertificatesMerged:
        local.crt: |
          -----BEGIN CERTIFICATE-----
          local
          -----END CERTIFICATE-----
        override.crt: |
          -----BEGIN CERTIFICATE-----
          local
          -----END CERTIFICATE-----
      clusterAcmName: clusterAcmName-local
      clusterApiURL: clusterApiURL-local
      clusterApiVersions:
        - clusterApiVersions-local
      clusterAppsDomain: clusterAppsDomain-local
      clusterBaseDomain: clusterBaseDomain-local
      clusterIsHub: false
      clusterKubeVersion: clusterKubeVersion-local
      clusterName: clusterName-local
      clusterRunsApps: false
      clusterServices:
        local:
          url: clusterServicesLocalUrl-local
        override:
          url: clusterServicesOverrideUrl-local
      clusterServicesExtracted: clusterServicesOverrideUrl-local
      clusterServicesMerged:
        local:
          url: clusterServicesLocalUrl-local
        override:
          url: clusterServicesOverrideUrl-local
      clusterType: clusterType-local
      customer: customer-local
      environment: environment-local
      environmentShort: environmentShort-local
      noProxy: noProxy-local
      proxy: proxy-local
      proxyIPs:
        - proxyIP-local1
        - proxyIP-local2
      proxyIPsArray:
        - IP: proxyIP-local1
        - IP: proxyIP-local2
      repoTargetRevision: repoTargetRevision-local
      repoURL: repoURL-local
      services:
        local:
          url: servicesLocaltUrl-local
        override:
          url: servicesOverrideUrl-local
      servicesExtracted: servicesOverrideUrl-local
      servicesMerged:
        local:
          url: servicesLocaltUrl-local
        override:
          url: servicesOverrideUrl-local
